### PR TITLE
cleanup cli/tests/default/e2e/test_test.py (was test_cli_test.py)

### PR DESCRIPTION
**Pull request by aryx**
Use better rule and target names, add comments,
move the test to the appropriate file, etc.

test plan:
cd cli; make e2e

### CODE REVIEW

File: cli/tests/default/e2e/test_test.py [hunk 1]
```diff
@@ -56,6 +94,7 @@ def test_cli_test_yaml_language(run_semgrep_in_tmp: RunSemgrep, snapshot):
     )
 
 
+# ??
```
Comment (amchiclet) [inline cli/tests/default/e2e/test_test.py]: This test checks that files with multiple suffixes (ie multiple dots) in the file names are included. (https://github.com/semgrep/semgrep/commit/6c07e59f16334e943bb2b0165ad3386e7909b12f)

File: cli/tests/default/e2e/test_test.py [hunk 2]
```diff
@@ -27,21 +42,44 @@ def test_cli_test_basic(run_semgrep_in_tmp: RunSemgrep, snapshot):
...
         output_format=OutputFormat.JSON,
+        assert_exit_code=1,
     )
     snapshot.assert_match(
         results,
         "results.json",
     )
 
 
+# ??
```
Comment (amchiclet) [inline cli/tests/default/e2e/test_test.py]: I think this is for testing that yaml files can also be targets. (https://github.com/semgrep/semgrep/commit/e2f48700d2a785154106b746fbc8c0167cd12c22)

**Review by amchiclet — APPROVED** [review]

File: cli/tests/default/e2e/test_test.py (hunk 1 above)
Comment (aryx) [reply cli/tests/default/e2e/test_test.py]: ok. Thx. Will fix the comment.

File: cli/tests/default/e2e/test_test.py (hunk 2 above)
Comment (aryx) [reply cli/tests/default/e2e/test_test.py]: Got it.

**Pull request merged**